testcases/OpTestXive.py

```diff
@@ -0,0 +1,210 @@
+            self.cv_HMC.vterm_run_command(self.console, "grubby --remove-args=xive* --update-kernel=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --args=xive=off --update-kernel=/boot/vmlinuz-`uname -r`")
+            time.sleep(5)
+            self.cv_HMC.vterm_run_command(self.console, "sync; sync;")
+            time.sleep(5)
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.restart_lpar()
+            self.cv_HMC.wait_login_prompt(self.console, username="root", password="passw0rd")
+            self.console = self.cv_HMC.get_console()
+            self.cv_HMC.vterm_run_command(self.console, "which stty && stty cols 300; which stty && stty rows 30;")
+            self.cv_HMC.vterm_run_command(self.console,"cat /proc/interrupts | grep -Eai 'xive'")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --remove-args=xive* --update-kernel=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "sync;sleep 10")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --args=xive=on --update-kernel=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "sleep 5; sync; sleep 5;")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.restart_lpar()
+            self.cv_HMC.wait_login_prompt(self.console, username="root", password="passw0rd")
+            self.console = self.cv_HMC.get_console()
+            self.cv_HMC.vterm_run_command(self.console,"cat /proc/interrupts | grep -Eai 'xive'")
+               '''
+    def xive_cpu_check(self): 
+        '''
+        This function will validate interrupts handled by CPU's.
+        Make sure while running this function xive should be on
+        '''
+        self.console = self.cv_HMC.get_console()
+        self.cv_HMC.vterm_run_command(self.console, "which stty && stty cols 300; which stty && stty rows 30;")
+        res = self.cv_HMC.vterm_run_command(self.console, "ip route list | grep default | awk '{print $5}'")
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts") 
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
+        time.sleep(2)
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
+        self.cv_HMC.vterm_run_command(self.console, "ppc64_cpu --cores-on")
+        self.cv_HMC.vterm_run_command(self.console, "ppc64_cpu --cores-on=1")
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
```

Comment:
Could not understand what this validation is to do with xive check, coz there is no explicit check for `xive ` here.
Can you explain a bit?

there can be multiple interrupts associated with single nw like below, 
xive host: (P9, smt=off, cores-on=1)
```
cat /proc/interrupts | grep net0 
123:         25  XIVE-IRQ 1028101 Edge      net0-tx-0
124:         79  XIVE-IRQ 1028102 Edge      net0-rx-1
125:          1  XIVE-IRQ 1028103 Edge      net0-rx-2
126:        244  XIVE-IRQ 1028104 Edge      net0-rx-3
127:          0  XIVE-IRQ 1028105 Edge      net0-rx-4
```
xics host: (P8: smt=off, cores-on=1)
```
/proc/interrupts | grep net0
412:    1007502  OPAL ICS 12303 Edge      net0-tx-0
413:   40214667  OPAL ICS 12304 Edge      net0-rx-1
414:     893219  OPAL ICS 12305 Edge      net0-rx-2
415:    3596368  OPAL ICS 12306 Edge      net0-rx-3
416:     727840  OPAL ICS 12307 Edge      net0-rx-4

```